packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -214,12 +214,231 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
 	}
 
 	public async getSnapshot(snapshotFetchOptions?: ISnapshotFetchOptions): Promise<ISnapshot> {
-		assert(this.hostPolicy.supportGetSnapshotApi !== true, "api should not be called yet");
-		// This is just temporary as this api is not yet enabled in service policies.
-		return this.fetchSnapshot(
-			this.hostPolicy.snapshotOptions,
-			snapshotFetchOptions?.scenarioName,
+		// Fetch full snapshot in case group is not specified in request.
+		let fetchFullSnapshotContents = true;
+		if (
+			snapshotFetchOptions?.loadingGroupIds !== undefined &&
+			snapshotFetchOptions.loadingGroupIds.length > 0
+		) {
+			fetchFullSnapshotContents = false;
+		}
+		// Don't consult cache if request is not for full snapshot.
+		const { snapshot } = await this.fetchSnapshot(
+			{
+				...snapshotFetchOptions,
+				fetchSource: fetchFullSnapshotContents
+					? snapshotFetchOptions?.fetchSource
+					: FetchSource.noCache,
+				loadingGroupIds: fetchFullSnapshotContents
+					? []
+					: snapshotFetchOptions?.loadingGroupIds,
```

Comment:
I really dislike that
1. The typing is still optional (i.e. undefined is allowed, and thus fetchSnapshot() and layers below needs to deal with it, even if it's never passed).
2. That this layer converts undefined -> [], and thus potentially changes semantics of what [] could mean.

My recommendations would stay the same - either do this kind of conversion at the lowers level (right before calling fetch()), or at the highest level (public API, also reflecting such change in typing - i.e. no undefined), and maintain exactly same semantics across all middle layers.
And remove fetchFullSnapshotContents - it's not needed, it should be calculated from the other arg,